app/modules/tcs34725.c

```diff
@@ -0,0 +1,371 @@
+	platform_i2c_send_byte(TCS34725_BUS_ID, TCS34725_COMMAND_BIT | reg);
+	platform_i2c_send_stop(TCS34725_BUS_ID);
+	
+	platform_i2c_send_start(TCS34725_BUS_ID);
+	platform_i2c_send_address(TCS34725_BUS_ID, TCS34725_ADDRESS, PLATFORM_I2C_DIRECTION_RECEIVER);
+	low = platform_i2c_recv_byte(TCS34725_BUS_ID, 1);
+	high = platform_i2c_recv_byte(TCS34725_BUS_ID, 0);
+	platform_i2c_send_stop(TCS34725_BUS_ID);
+	
+	return (high << 8) | low;;
+}
+/**************************************************************************/
+/*!
+		@brief	Finishes enabling the device
+*/
+/**************************************************************************/
+uint8_t tcs34725EnableDone(lua_State* L)
+{
+	dbg_printf("Enable second bit\n");
+	tcs34725Write8(TCS34725_ENABLE, TCS34725_ENABLE_PON | TCS34725_ENABLE_AEN);
+
+	/* Ready to go ... set the initialised flag */
+	_tcs34725Initialised = true;
+	
+	/* This needs to take place after the initialisation flag! */
+	tcs34725SetIntegrationTime(TCS34725_INTEGRATIONTIME_2_4MS, L);
+	tcs34725SetGain(TCS34725_GAIN_60X, L);
+	return 0;
+}
+
+/**************************************************************************/
+/*!
+		@brief	Enables the device
+*/
+/**************************************************************************/
+uint8_t tcs34725Enable(lua_State* L)
+{
+	dbg_printf("Enable first bit\n");
+	tcs34725Write8(TCS34725_ENABLE, TCS34725_ENABLE_PON);
+	// Start a timer to wait TCS34725_EN_DELAY before calling tcs34725EnableDone
+	os_timer_disarm (&tcs34725_timer);
+	os_timer_setfn (&tcs34725_timer, (os_timer_func_t *)tcs34725EnableDone, L);
+	os_timer_arm (&tcs34725_timer, TCS34725_EN_DELAY, 0); // trigger callback when readout is ready
+
+	return 0;
+}
+
+/**************************************************************************/
+/*!
+		@brief	Disables the device (putting it in lower power sleep mode)
+*/
+/**************************************************************************/
+uint8_t tcs34725Disable(lua_State* L)
+{
+	/* Turn the device off to save power */
+	uint8_t reg = 0;
+	reg = tcs34725Read8(TCS34725_ENABLE);
+	tcs34725Write8(TCS34725_ENABLE, reg & ~(TCS34725_ENABLE_PON | TCS34725_ENABLE_AEN));
+
+	return 0;
+}
+
+/**************************************************************************/
+/*!
+		@brief	Initialises the I2C block
+*/
+/**************************************************************************/
+uint8_t tcs34725Init(lua_State* L)
+{
+	int lua_dbg_cb_ref;
+	uint8_t id = 0;
+	uint8_t sda;
+	uint8_t scl;
+	uint8_t config;
+	uint8_t ack;
+
+	if (!lua_isnumber(L, 1) || !lua_isnumber(L, 2)) {
+		return luaL_error(L, "wrong arg range");
+	}
+	sda = luaL_checkinteger(L, 1);
+	scl = luaL_checkinteger(L, 2);
+	platform_i2c_setup(TCS34725_BUS_ID, sda, scl, PLATFORM_I2C_SPEED_SLOW);
+	
+	/* Make sure we have the right IC (0x44 = TCS34725 and TCS34721) */
+	id = tcs34725Read8(TCS34725_ID);
+	dbg_printf("id: %x\n",id);
+	if (id != 0x44) {
+		luaL_error(L, "No TCS34725 found.");
+		return 0;
```

Comment:
Here and below: `luaL_error()` implicitly leaves the function, see [`luaL_error`](http://pgl.yoyo.org/luai/i/luaL_error).
You could use the `return luaL_error()` pattern for clarity.